var/spack/repos/builtin/packages/esmf/package.py

```diff
@@ -0,0 +1,256 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+import os
+
+
+class Esmf(Package):
+    """The Earth System Modeling Framework (ESMF) is high-performance, flexible
+    software infrastructure for building and coupling weather, climate, and
+    related Earth science applications. The ESMF defines an architecture for
+    composing complex, coupled modeling systems and includes data structures
+    and utilities for developing individual models."""
+
+    homepage = "https://www.earthsystemcog.org/projects/esmf/"
+    url      = "http://www.earthsystemmodeling.org/esmf_releases/non_public/ESMF_7_0_1/esmf_7_0_1_src.tar.gz"
+
+    version('7.0.1', 'd3316ea79b032b8fb0cd40e5868a0261')
+
+    variant('mpi',     default=True,  description='Build with MPI support')
+    variant('lapack',  default=True,  description='Build with LAPACK support')
+    variant('netcdf',  default=True,  description='Build with NetCDF support')
+    variant('pnetcdf', default=True,  description='Build with pNetCDF support')
+    variant('xerces',  default=True,  description='Build with Xerces support')
+    variant('pio',     default=True,  description='Enable ParallelIO support')
+    variant('debug',   default=False, description='Make a debuggable version of the library')
+
+    # Required dependencies
+    depends_on('mpi', when='+mpi')
+    depends_on('zlib')
+    depends_on('libxml2')
+    # depends_on('perl', type='test')  # TODO: Add a test deptype
+
+    # Optional dependencies
+    depends_on('lapack@3:', when='+lapack')
+    depends_on('netcdf@3.6:', when='+netcdf')
+    depends_on('netcdf-fortran@3.6:', when='+netcdf')
+    depends_on('parallel-netcdf@1.2.0:', when='+pnetcdf')
+    depends_on('xerces-c@3.1.0:', when='+xerces')
+
+    # NOTE: ESMF cannot be installed with GCC 6. It uses constructs that
+    # are no longer valid in GCC 6. It should install properly with GCC 5
+
+    # NOTE: ESMF cannot be installed with MVAPICH2. It results in undefined
+    # references that cause `make install` to crash.
+
+    def url_for_version(self, version):
+        return "http://www.earthsystemmodeling.org/esmf_releases/non_public/ESMF_{0}/esmf_{0}_src.tar.gz".format(version.underscored)
+
+    def install(self, spec, prefix):
+        # Installation instructions can be found at:
+        # http://www.earthsystemmodeling.org/esmf_releases/last_built/ESMF_usrdoc/node9.html
+
+        # Unset any environment variables that may influence the installation
+        for var in os.environ:
+            if var.startswith('ESMF_'):
+                os.environ.pop(var)
```

Comment:
Currently we only proactively remove things that we know could cause issues.  There is probably a case to be made for just wiping the entire environment and whitelisting instead of blacklisting, but it won't work on Cray.  I'd be enthused to test something more aggressive out after we have package testing.